bokehjs/src/coffee/common/pgrid.coffee

```diff
@@ -0,0 +1,109 @@
+$ = require "jquery"
+_ = require "underscore"
+Backbone = require "backbone"
+build_views = require "./build_views"
+ContinuumView = require "./continuum_view"
+HasProperties = require "./has_properties"
+{logger} = require "./logging"
+widget = require "phosphor-widget"
+gridpanel = require "phosphor-gridpanel"
+messaging = require "phosphor-messaging"
+
+
+class PGridView extends ContinuumView
+  #className: ""
+
+  initialize: (options) ->
+    super(options)
+    window.box = this
+    @views = {}
+    
+    @panel = new gridpanel.GridPanel()
+    @panel.node.style.width = '100%'
+    @panel.node.style.height = '100%'
+    @panel.node.style.position = 'absolute'
+    @el.appendChild(@panel.node)
+    # simulate widget.attachWidget(@panel, @el), but allow @el to be unbound to DOM
+    messaging.sendMessage(@panel, widget.MSG_AFTER_ATTACH)
+    
+    @build_children()
+    @bind_bokeh_events()
+    @render()
+    return this
+
+  bind_bokeh_events: () ->
+    @listenTo(@model, 'change:children', @build_children)
+    @listenTo(@model, 'change', @render)
+    @listenTo(@model, 'destroy', @remove)
+    window.addEventListener('resize', () -> box.panel.update())
+
+  build_children: () ->
+    # Get views
+    child_models = []
+    for row in @mget('children')
+      for child in row
+        if not child?
+          continue
+        child_models.push(child)
+    build_views(@views, child_models, {})
+    # Populate grid
+    @panel.clearChildren()
```

Comment:
You don't need to clear children here. It will be done automatically when you assign the children below.